src/loan.rs

```diff
@@ -888,14 +900,50 @@ impl Lender0 {
     /// Interpret a loan request and performs lender logic.
     ///
     /// rate is expressed in usdt sats per btc, i.e. rate = 1 BTC / USDT
+    #[deprecated(note = "Use interpret_loan_request instead")]
     pub async fn interpret<R, C, CS, CF>(
         self,
         rng: &mut R,
         secp: &Secp256k1<C>,
         coin_selector: CS,
         loan_request: LoanRequest,
+        timelock: u32,
         rate: u64,
     ) -> Result<Lender1>
+    where
+        R: RngCore + CryptoRng,
+        C: Verification + Signing,
+        CS: FnOnce(Amount, AssetId) -> CF,
+        CF: Future<Output = Result<Vec<Input>>>,
+    {
+        self.interpret_loan_request(
+            rng,
+            secp,
+            coin_selector,
+            rate,
+            loan_request.fee_sats_per_vbyte,
+            (
+                loan_request.collateral_amount,
+                loan_request.collateral_inputs,
+            ),
+            (loan_request.borrower_pk, loan_request.borrower_address),
+            timelock,
+        )
+        .await
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    pub async fn interpret_loan_request<R, C, CS, CF>(
```

Comment:
Hmm, what do you think of naming this more from the perspective of the state machine that is being modeled, like `finalize_loan` or some other verb that is appropriate?

Reply:
I went for `build_loan_transaction` since that's what happening internally. Didn't want to say "finalize" since the signatures are missing.